engine/class_modules/monk/sc_monk.cpp

```diff
@@ -7313,8 +7326,12 @@ void monk_t::init_spells()
 
   shared.rushing_jade_wind = _priority( talent.windwalker.rushing_jade_wind, talent.brewmaster.rushing_jade_wind );
 
-  shared.teachings_of_the_monastery =
-      _priority( talent.windwalker.teachings_of_the_monastery, talent.mistweaver.teachings_of_the_monastery );
+  if ( talent.windwalker.teachings_of_the_monastery->ok() )
+    shared.teachings_of_the_monastery = talent.windwalker.teachings_of_the_monastery;
+  else if ( baseline.mistweaver.teachings_of_the_monastery->ok() )
+    shared.teachings_of_the_monastery = baseline.mistweaver.teachings_of_the_monastery;
+  else
+    shared.teachings_of_the_monastery = spell_data_t::not_found();
```

Comment:
If you're not comfortable writing a new equivalent function to `_priority` that is more friendly to multiple parameter types, let me know and I can put something together.

This feels gross, even I'm generally trying to get away from shared spell pointers.

Reply:
sure, no idea how to rewrite `_priority` tbh